File: app/src/main/java/party/qwer/iris/PageRenderer.java
// ConfigPageDocumentProvider.java
package party.qwer.iris;

import java.io.IOException;

public class PageRenderer {
    public static String renderDashboard() throws IOException {
        String html = AssetManager.readFile("dashboard.html");
        html = html.replace("CURRENT_WEB_ENDPOINT", Configurable.INSTANCE.getWebServerEndpoint());
        html = html.replace("CURRENT_BOT_NAME", Configurable.INSTANCE.getBotName());
        html = html.replace("CURRENT_DB_RATE", String.valueOf(Configurable.INSTANCE.getDbPollingRate()));
        html = html.replace("CURRENT_SEND_RATE", String.valueOf(Configurable.INSTANCE.getMessageSendRate()));
        html = html.replace("CURRENT_BOT_PORT", String.valueOf(Configurable.INSTANCE.getBotSocketPort()));
        return html;
    }
}

File: app/src/main/java/party/qwer/iris/PageRenderer.kt
// ConfigPageDocumentProvider.java
package party.qwer.iris

import party.qwer.iris.Configurable.botName
import party.qwer.iris.Configurable.botSocketPort
import party.qwer.iris.Configurable.dbPollingRate
import party.qwer.iris.Configurable.messageSendRate
import party.qwer.iris.Configurable.webServerEndpoint
import java.io.IOException

object PageRenderer {
    @Throws(IOException::class)
    fun renderDashboard(): String {
        var html = AssetManager.readFile("dashboard.html")
        html = html.replace("CURRENT_WEB_ENDPOINT", webServerEndpoint)
        html = html.replace("CURRENT_BOT_NAME", botName)
        html = html.replace("CURRENT_DB_RATE", dbPollingRate.toString())
        html = html.replace("CURRENT_SEND_RATE", messageSendRate.toString())
        html = html.replace("CURRENT_BOT_PORT", botSocketPort.toString())
        return html
    }
}